ci/setup-and-build.ps1

```diff
@@ -0,0 +1,164 @@
+            pushd "Plugins"
+            Start-Process -Wait -PassThru -NoNewWindow -FilePath "git" -ArgumentList @(`
+                "clone", `
+                "git@github.com:spatialos/UnrealGDK.git", `
+                "--depth 1", `
+                "-b $gdk_branch_name"
+            )
+            popd
+        popd
+    Finish-Event "clone-gdk-plugin" "build-gdk-third-person-shooter-:windows:"
+
+    Start-Event "set-up-gdk-plugin" "build-gdk-third-person-shooter-:windows:"
+        pushd $gdk_path
+            # Set the required variables for the GDK's setup script to use
+            $msbuild_exe = "${env:ProgramFiles(x86)}\MSBuild\14.0\bin\MSBuild.exe"
+
+            # Invoke the setup script
+            &"$($gdk_path)\ci\setup-gdk.ps1"
+        popd
+    Finish-Event "set-up-gdk-plugin" "build-gdk-third-person-shooter-:windows:"
+
+    # Fetch the version of Unreal Engine we need
+    pushd "$($gdk_path)/ci"
+        $unreal_version = Get-Content -Path "unreal-engine.version" -Raw
+        Write-Log "Using Unreal Engine version: $unreal_version"
+    popd
+
+    Start-Event "download-unreal-engine" "build-gdk-third-person-shooter-:windows:"
+        ## Create an UnrealEngine directory if it doesn't already exist
+        New-Item -Name "UnrealEngine" -ItemType Directory -Force
+
+        pushd "UnrealEngine"
+            Write-Log "Downloading the Unreal Engine artifacts from GCS"
+            $gcs_unreal_location = "$($unreal_version).zip"
+
+            $gsu_proc = Start-Process -Wait -PassThru -NoNewWindow "gsutil" -ArgumentList @(`
+                "cp", `
+                "gs://$($gcs_publish_bucket)/$($gcs_unreal_location)", `
+                "$($unreal_version).zip" `
+            )
+            if ($gsu_proc.ExitCode -ne 0) {
+                Write-Log "Failed to download Engine artifacts. Error: $($gsu_proc.ExitCode)"
+                Throw "Failed to download Engine artifacts"
+            }
+
+            Write-Log "Unzipping Unreal Engine"
+            $zip_proc = Start-Process -Wait -PassThru -NoNewWindow "7z" -ArgumentList @(`
+            "x", `
+            "$($unreal_version).zip" `
+            )
+            if ($zip_proc.ExitCode -ne 0) {
+                Write-Log "Failed to unzip Unreal Engine. Error: $($zip_proc.ExitCode)"
+                Throw "Failed to unzip Unreal Engine."
+            }
+        popd
+    Finish-Event "download-unreal-engine" "build-gdk-third-person-shooter-:windows:"
+
+    Start-Event "install-unreal-engine-prerequisites" "build-gdk-third-person-shooter-:windows:"
+        # This runs an opaque exe downloaded in the previous step that does *some stuff* that UE needs to occur.
+        # Trapping error codes on this is tricky, because it doesn't always return 0 on success, and frankly, we just don't know what it _will_ return.
+        Start-Process -Wait -PassThru -NoNewWindow -FilePath "$($unreal_path)\Engine\Extras\Redist\en-us\UE4PrereqSetup_x64.exe" -ArgumentList @(`
```

Comment:
This uses $unreal_path which is only set below.

Reply:
Oh dear lord yes